xmake/rules/qt/install/windows.lua

```diff
@@ -0,0 +1,90 @@
+--!A cross-platform build utility based on Lua
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+-- Copyright (C) 2015-2020, TBOOX Open Source Group.
+--
+-- @author      ruki
+-- @file        macosx.lua
+--
+
+-- imports
+import("core.theme.theme")
+import("core.base.option")
+import("core.project.config")
+import("core.project.depend")
+import("detect.sdks.find_vstudio")
+import("private.utils.progress")
+
+-- install application package for windows
+function main(target, opt)
+
+    local installfile = path.join(target:installdir(), "bin", target:basename() .. ".exe")
+
+    -- need re-generate this app?
+    local targetfile = target:targetfile()
+    local dependfile = target:dependfile(targetfile)
+    local dependinfo = option.get("rebuild") and {} or (depend.load(dependfile) or {})
+    if not depend.is_changed(dependinfo, {lastmtime = os.mtime(dependfile)}) then
+        return
+    end
+
+    -- trace progress info
+    -- progress.show(opt.progress, "${color.build.target}generating.qt.app %s.app", target:basename())
+
+    -- get qt sdk
+    local qt = target:data("qt")
+
+    -- get windeployqt
+    local windeployqt = path.join(qt.bindir, "windeployqt.exe")
+    assert(os.isexec(windeployqt), "windeployqt.exe not found!")
+
+    -- find qml directory
+    local qmldir = nil
+    for _, sourcebatch in pairs(target:sourcebatches()) do
+        if sourcebatch.rulename == "qt.qrc" then
+            for _, sourcefile in ipairs(sourcebatch.sourcefiles) do
+                qmldir = find_path("*.qml", path.directory(sourcefile))
+                if qmldir then
+                    break
+                end
+            end
+        end
+    end
+
+    -- do deploy
+    local vs_studio = find_vstudio()
+
+    -- Find Visual Studio
+    local installDir = path.join(vs_studio["2019"].vcvarsall[target:arch()].VSInstallDir, "VC")
+    os.addenv("VCINSTALLDIR", installDir)
```

Comment:
Oh, maybe you need to submit this pr to the toolchain branch, because I recently improved how to get vcvars data on the toolchain branch, and I will merge this branch in the last few days.

You can try this on toolchain branch.

In addition, we don’t need to call find_vstudio every time to find vs.

```lua
import("core.tool.toolchain")

local envs = nil
local msvc = toolchain.load("msvc", {plat = target:plat(), arch = target:arch()})
if msvc then
    local vcvars = msvc:config("vcvars")
    if vcvars and vcvars.VSInstallDir then
        envs = {VCINSTALLDIR = path.join(vcvars.VSInstallDir, "VC")}
    end
end

os.vrunv(windeployqt, argv, {envs = envs})
```

Reply:
>     os.addenv("VCINSTALLDIR", installDir)

This will affect the entire global environment, we only need to set environment variables to os.vrunv

```lua
os.vrunv(windeployqt, argv, {envs = envs})
```

Reply:
> Oh, maybe you need to submit this pr to the toolchain branch, because I recently improved how to get vcvars data on the toolchain branch, and I will merge this branch in the last few days.

Or please wait a while, I first merge the toolchain branch to dev, and then you try it again.

Reply:
I have merged toolchain to dev, you can run `git rebase` and try it again.